docs/activityadvanced.md

```diff
@@ -0,0 +1,137 @@
+# Advanced steps
+
+Glad to see you are still here!
+
+These are some additional exercises you can do on top of the
+[previous activity steps](https://microbit-carlos.github.io/microbit-bird-activity/activity/).
+
+
+## Expanding the states
+
+In `main.py` we've checked the bird state using the `bird.current_state()`
+function:
+
+```python
+while True:
+    bird_state = bird.current_state()
+    if bird_state == "chill":
+        display.show(Image.HAPPY)
+```
+
+How does that work? Let's look at the `bird.py` file, and find the
+`current_state()` function:
+
+```python
+def current_state():
+    # Check for motion
+    if __mb.accelerometer.was_gesture('shake'):
+        return 'angry'
+
+    __mb.sleep(10)   # Ensure run_every/gestures have a chance to run
+    return "chill"
+```
+
+<p class="exercise">
+Exercise: <br>
+Birds can be noisy, but they don't like loud sounds themselves!
+The micro:bit has a microphone, and we can use it to detect loud noises,
+so let's create a new state: "startled".
+<br>
+Let's add another if statement inside the
+<code class="language-plaintext highlighter-rouge">current_state()</code>
+function inside the
+<code class="language-plaintext highlighter-rouge">bird.py</code>  file.
+So, if there was a "loud" event, the function can return the `startled` state.
+<br>
+You can find more info about the microphone in the <a href="https://microbit-micropython.readthedocs.io/en/v2-docs/microphone.html" target="_blank">Microphone documentation</a>.
+</p>
+
+[Solution](https://github.com/microbit-carlos/microbit-bird-activity/tree/main/solutions/4-bird-startled.md)
+
+Now make a noise, like a clap, near the microphone and watch your bird get
+startled!
+
+Hint: If the conference area is too loud, you can blow on the microphone.
```

Comment:
Can't they also set the threshold for loud?

Reply:
Yes, that's also a good tip! Added in 21ad6bda79cb278392ffcc3482dd4da27274a3cb